bigtable/use-cases/fraudDetection/src/main/java/bigtable/fraud/utils/TransactionDetails.java

```diff
@@ -116,6 +120,87 @@ public String toCommaSeparatedString() {
     return "Transaction id: " + transactionID + ", isFraud: " + isFraud;
   }
 
+  /**
+   * @param row the row result read from Cloud Bigtable.
+   * @param duration the duration in milliseconds to go back in time for.
+   * @return the last transaction details in the last "duration" for this
+   * customer.
+   */
+  public ArrayList<TransactionDetails> getLastTransactions(
+      final Result row, final long duration)
+      throws IllegalAccessException {
+    ArrayList<TransactionDetails> lastTransactions = new ArrayList<>();
+    String[] headers = getHeaders();
+
+    // Create ArrayList that will hold the cells when we read from CBT and
+    // ignore the first element because it will hold the rowKey, and we
```

Comment:
turns out i was mistaken and row key is two words not one, you don't have to go and change it in all the places if it's a pain but just a heads up